examples/assignment-1/src/main/java/ro/utcn/spet/example/a1/service/QtagManagementService.java

```diff
@@ -0,0 +1,45 @@
+package ro.utcn.spet.example.a1.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+
+import ro.utcn.spet.example.a1.entity.QTag;
+import ro.utcn.spet.example.a1.repository.RepositoryFactory;
+
+import java.util.Collections;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class QtagManagementService {
+    private final RepositoryFactory repositoryFactory;
+
+    // Transactional methods ensure that the code contained inside is run in a transaction, which is committed
+    // when the methods returns and is rolled-back when an exception is thrown
+    // http://www.codingpedia.org/jhadesdev/how-does-spring-transactional-really-work/
+    @Transactional
+    public List<QTag> qTagList() {
+        List<QTag> myQuestions = repositoryFactory.createQtagRepository().findAll();
+        Collections.reverse(myQuestions);
+
+        return myQuestions;
+
+    }
+
+
+
+    @Transactional
+    public QTag insert ( int id_q,int id_t) {
+        return repositoryFactory.createQtagRepository().save(new QTag(id_q,id_t));
+    }
+
+    @Transactional
+    public List<QTag> findidq(int id_t) {
```

Comment:
Overall inconsistent naming. In Java, you should **always** use camel case for method names and parameters.

For example: `tagId` instead of `id_t`, `questionId` instead of `id_q`, `findQuestionsByTag` instead of `findidq`.

Reply:
It must have been hard to look over such a poor written code...I am really sorry...usually I focus so hard on how to implement and fix errors that I name everything pretty random. I will try to fix it in the future.

Reply:
It was not that bad for me to read :) But it is very good if you can focus in the future to improve this. Consistent and meaningful naming of identifiers is one of the most important things that you can learn as a developer, as it will help both other people and yourself to understand the code better.